src/main/java/io/cdap/plugin/gcp/bigquery/sqlengine/builder/BigQueryWindowsAggregationSQLBuilder.java

```diff
@@ -0,0 +1,135 @@
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+/**
+ * Helper class used to generate BigQuery SQL Statements for Select statements.
+ */
+public class BigQueryWindowsAggregationSQLBuilder extends BigQueryBaseSQLBuilder {
+  private static final String ROW_NUM_PREFIX = "rn_";
+  private final WindowAggregationDefinition windowAggregationDefinition;
+  private static final Logger LOG = LoggerFactory.getLogger(BigQueryWindowsAggregationSQLBuilder.class);
+  private final String source;
+  private final String sourceAlias;
+  private final String rowNumColumnAlias;
+  private final StringBuilder builder;
+
+  public BigQueryWindowsAggregationSQLBuilder(WindowAggregationDefinition windowAggregationDefinition,
+                                              String sourceExpression,
+                                              String sourceAlias) {
+    this(windowAggregationDefinition,
+         sourceExpression,
+         sourceAlias,
+         ROW_NUM_PREFIX + BigQuerySQLEngineUtils.newIdentifier());
+
+  }
+
+  protected BigQueryWindowsAggregationSQLBuilder(WindowAggregationDefinition windowAggregationDefinition,
+                                                 String source,
+                                                 String sourceAlias,
+                                                 String rowNumColumnAlias) {
+    this.windowAggregationDefinition = windowAggregationDefinition;
+    this.source = source;
+    this.sourceAlias = sourceAlias;
+    // This is the alias used to store the row number value. Format is "rn_<uuid>"
+    this.rowNumColumnAlias = rowNumColumnAlias;
+    this.builder = new StringBuilder();
+  }
+
+  @VisibleForTesting
+  protected String getSelectedFields(Map<String, Expression> selectedFields) {
+    List<Expression> e = new ArrayList<>(selectedFields.values());
+    return getPartitionFields(e);
+  }
+
+  @Override
+  public String getQuery() {
+    String overClause = getOverClause();
+    String query = SELECT + getSelectedFields(windowAggregationDefinition.getSelectExpressions()) + SPACE + overClause +
+      SPACE + AS + sourceAlias;
+    LOG.debug("Query is " + query);
+    return query;
+  }
+  
+  private String getOverClause() {
+    return OVER +
+      OPEN_GROUP + SPACE + PARTITION_BY + SPACE +
+      getPartitionFields(windowAggregationDefinition.getPartitionExpressions()) + SPACE
+      + getOrderByFields(windowAggregationDefinition.getOrderByExpressions()) + SPACE +
+      getWindowFrameDefinition(windowAggregationDefinition) + SPACE + CLOSE_GROUP;
+  }
+
+  private String getWindowFrameDefinition(WindowAggregationDefinition windowAggregationDefinition) {
+    String def = "";
+    if (windowAggregationDefinition.getWindowFrameType() == WindowAggregationDefinition.WindowFrameType.NONE) {
+      return def;
+    }
+    if (windowAggregationDefinition.getWindowFrameType() == WindowAggregationDefinition.WindowFrameType.RANGE) {
+      def = def + "RANGE";
```

Comment:
Make these also static final values (such as AND, SPACE, ETC)

Reply:
DONE